server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -518,6 +524,14 @@ public synchronized IndexShard createShard(
                 remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY);
             }
 
+            TranslogFactory translogFactory = this.indexSettings.isRemoteTranslogStoreEnabled()
```

Comment:
As we are checking for indexSettings, we will provide `RemoteBlobStoreInternalTranslogFactory` for primaries as well as replicas, is this expected?

Reply:
Yes, created a [backlog item](https://github.com/opensearch-project/OpenSearch/issues/5640) for this.

Reply:
Changed this to use `InternalTranslogFactory` for now in replicas as the bootup of replica was failing  .